src/madness/world/parsec.h

```diff
@@ -15,12 +15,21 @@
 #include <iostream>
 
 namespace madness{
-  extern "C"{
-
-    extern const parsec_task_class_t madness_parsec_tc;
-    extern parsec_taskpool_t madness_parsec_tp;
-  }
-
+    class PoolTaskInterface;
+
+    class ParsecRuntime {
+    private:
+        static parsec_context_t *context;
```

Comment:
Why are they all static? The parsec_runtime object is already static and even if it wasn't, I don't see how there can a constructor that intiializes static members, esp. since that constructor asserts if context is already set.

Reply:
schedule(), test() and task() are called from  static methods in ThreadPool, that makes them become static, and then they refer the taskpool, that needed to become static too.

Reply:
Sorry, let me clarify: The ThreadPool has a static member parsec_runtime. There is no need for the member variables and functions of the ParsecRuntime to be static, too. They can be called like any member function on the static parsec_runtime member of ThreadPool. 

The other question would be: can there be more than one ThreadPool? And if so, who has ownership of the parsec context? (i.e., who is responsible for its destruction?) If there can be more than one over the lifetime of an application, the ctor ParsecRuntime would assert on the second creation of the ThreadPool...

Reply:
@devreal `ThreadPool` is a singleton. So its methods (in this particular design) became static functions